drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -336,6 +336,57 @@ def data_collection_features
     end.compact
   end
 
+  def occurrence_point_forms
+    # All instances for Occurrence Point forms, including inactive/retired, so we can check for legacy data.
+    # Draft forms should never have legacy data, but let's include them just in case.
+    base_scope = Hmis::Form::Instance.with_role(:OCCURRENCE_POINT)
+
+    # All possible form identifiers used for Occurrence Point collection
+    occurrence_point_identifiers = base_scope.pluck(:definition_identifier).uniq
+
+    # Get the latest version of each form definition so we can check for legacy data
+    # todo @martha - edge case with drafts
```

Comment:
There is an edge case here with draft forms that I haven't really decided is worth going down the rabbit hole for. What do you think?
- Suppose definition identifier 'foo' has two versions. Definition 0 is published and has question 'baz'. Definition 1, in draft, has the 'baz' question removed. 
- `latest_versions` returns definition 1.
- This wouldn't impact the `best_instance` chosen on line 354, since that is chosen by identifier.
- Suppose no `best_instance` is found for this enrollment, now we look for `has_any_data`.
- Suppose the enrollment has legacy data for 'baz'. We would not catch that, since `has_any_data` is walking definition nodes on definition 1, which lacks the 'baz' question.

Hm 🦆 As I type this out I'm realizing this is not a specific problem with draft forms. It could happen anytime a previous version of a form collected different data than a later version. It just seems especially egregious when the latest version is a draft. 

I don't think it seems at all worthwhile to iterate over *all* previous form definitions (let me know if you agree). 

But maybe for draft forms specifically, we should walk definition nodes on the published version, if one exists.

Reply:
Ok, I made that change. Let me know how you think it looks!

Reply:
Looks good! Left a comment about drafts https://github.com/greenriver/hmis-warehouse/pull/4877#discussion_r1842134995